### PR TITLE
[SPARK-40818][CONNECT] Add Intersect to Connect proto and DSL

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR supports `Intersect` to Connect proto and DSL.

### Why are the changes needed?

To improve proto API coverage.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unittest was added.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @amaliujia @zhengruifeng PTAL when you find some time.

File: connector/connect/src/main/protobuf/spark/connect/relations.proto [hunk 1]
```diff
@@ -41,6 +41,7 @@ message Relation {
     Aggregate aggregate = 9;
     SQL sql = 10;
     LocalRelation local_relation = 11;
+    Intersect intersect = 12;
```
Comment (HyukjinKwon) [inline connector/connect/src/main/protobuf/spark/connect/relations.proto:44]: This conflicts with some other PRs. I will change this if other PRs are merged first.

**Comment by zhengruifeng** [comment]
you should regenerate the pb files now:
```
Start checking the generated codes in pyspark-connect.
RUN: /__w/spark/spark/connector/connect/dev/generate_protos.sh /tmp/tmpfq9npbm2
Different files: ['relations_pb2.py', 'relations_pb2.pyi']
Generated files for pyspark-connect are out of sync! Please run ./connector/connect/dev/generate_protos.sh
Error: Process completed with exit code 255.
```

File: python/pyspark/sql/connect/proto/relations_pb2.pyi [hunk 2]
```diff
@@ -70,6 +70,7 @@ class Relation(google.protobuf.message.Message):
     AGGREGATE_FIELD_NUMBER: builtins.int
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/proto/relations_pb2.pyi:70]: I hope we can get this of these codes in the main repo ..

**Comment by amaliujia** [comment]
oops sorry for it: this PR duplicates with https://github.com/apache/spark/pull/38166. 

Basically people wants to merge Union, Intersect and Except with `is_all` and `by_name` together.

File: connector/connect/src/main/protobuf/spark/connect/relations.proto (hunk 1 above)
Comment (grundprinzip) [reply connector/connect/src/main/protobuf/spark/connect/relations.proto:44]: Wait, don't we have a Union type that does this (e.g. from the set operations?)
@amaliujia ?

**Comment by HyukjinKwon** [comment]
Oops, sorry guys. Closing

**Pull request closed without merging**